optaplanner-core/src/test/java/org/optaplanner/core/api/solver/SolverManagerTest.java

```diff
@@ -240,4 +245,86 @@ public void skipAhead() throws ExecutionException, InterruptedException {
         assertTrue(eventCount.get() < 4);
     }
 
+    /**
+     * In order to effectively test the terminateEarly() and close() methods there had to be a way how to make the job
+     * status change deterministic while solving jobs on a separate thread. The test waits for a status change in
+     * a loop and time-outs when it doesn't reach it's expected status.
+     */
+    @Test
+    public void terminateEarlyAndClose() throws InterruptedException {
```

Comment:
If this test fails (as it reproduces an issue), let's mark it as ignored with a link to that issue and get this merged. Once the issue is fixed, it will be enabled.

Reply:
This timeout approach is not race condition free. If we activate this test, it will cause randomly failing unit tests (which is never acceptable). I am going to refactor this to use Latches.